### DISCUSSION TITLE
Some notes about Geany 2.1 in LMC 21.3

### DISCUSSION THREAD

**Discussion by ineuw** [geany/geany#3920] [General]
I am sure that these issues relate to not having enough help to update everything, but the best I can do, is note them for the sake of record keeping.

* The Help link https://www.geany.org/manual/2.1/index.html in Help is 404.
* The home page https://www.geany.org/ promotes version 2.0.
* Please add a link to the HTML version of the manual in the Help list.
* Do the dropdown search and replace lists work between sessions to be accessed?
* I created a copy of a colorschema as ineuw.conf in the local installation but it's not listed.
* Is v2.1 using a single colorschema folder for user and superuser? 
* Can I disable bracket highlights? Where?

**Comment by b4n** [comment]
That's quite a few questions, but…

> * The Help link https://www.geany.org/manual/2.1/index.html in Help is 404.
> * The home page https://www.geany.org/ promotes version 2.0.

Geany 2.1 isn't released yet (don't trust the milestone date, it's not closed yet)

> * Please add a link to the HTML version of the manual in the Help list.

Where do you mean?

> * I created a copy of a colorschema as ineuw.conf in the local installation but it's not listed.
> * Is v2.1 using a single colorschema folder for user and superuser?

There is the system colorscheme directory (usually */usr/share/geany/colorschemes/*, or the same under */usr/local/* if you installed manually), and the one in the Geany configuration, *~/.config/geany/colorschemes/*.

Unrelated note: I advise against running Geany as root.

> * Can I disable bracket highlights? Where?

You can make it look like the default style using *Tools -> Configuration files -> filetypes.common* and set `brace_good=default` in the `[named_styles]` section.

**Comment by elextr** [comment]
> Do the dropdown search and replace lists work between sessions to be accessed?

Not totally sure what you mean, if you mean the dropdown lists on the entrys in the search/replace dialog get saved and restored when Geany is shutdown and restarted, then no.

> > Is v2.1 using a single colorschema folder for user and superuser?

> There is the system colorscheme directory (usually /usr/share/geany/colorschemes/, or the same under /usr/local/ if you installed manually)

Or `$prefix/geany/colorschemes` if you built it somewhere else

> Unrelated note: I advise against running Geany as root.

+100000000000000000000000000000000  Ubuntu (and descendents) broken sudo screws up which $home is used, so it depends on which distro you use what the answer to your question is, but its not different to 2.0 if thats what you mean.

> Can I disable bracket highlights? Where?

See https://github.com/geany/geany/issues/3555

**Comment by ineuw** [comment]
To all responders, my belated thanks, but I needed time for exploring Geany further for my requirements. I use Geany in Linux Mint Cinnamon 21 + 22. My only question remains: why not use Geany with "sudo" ? I am a home user and not part of a network except my own LAN. Please elaborate only if it's relevant, otherwise consider the topic closed.

**Comment by elextr** [comment]
The sudo (assuming root) problem is that files written by Geany running as root get the root ownership, the files that are edited, but also the status files saved in your own directory, things like config, session, the socket etc.   Clearly that is a problem since it is likely that then Geany run as your normal user may not be able to overwrite or delete or otherwise change any file that has now been owned by root.

**Comment by b4n** [comment]
There are two issues with using Geany as root, and via sudo:

* the general issue of running a complex GUI program as root, where mistakes are easier to make, and especially one with plugins of various quality. And also note that it probably hasn't been tested in that very situation.
* the sudo-specific issue that as @elextr mentioned: Ubuntu and derivative usually have a sudo configuration that leads to writing root-owned files in the calling user's homedir. I don't know if there's any reason for that configuration but it's very problematic in many cases. Note that it's not inherent to sudo, Debian work right in this respect.

Once you know this, do as you feel, but remember it's not a use case we support or recommend.

If editing root-owned files in Geany is something people want badly enough, *somebody* could research what it would take to support GVFS' `admin://` URI scheme.

**Reply by ineuw** [reply]
Thanks for the clarifications. Does this apply to opening a file as root from Nemo file navigator in Linux Mint Cinnamon 22? I am referring to the /etc/fstab file.

**Reply by elextr** [reply]
You should check it for yourself, none of us can see your system, we do not necessarily know any particular system, we don't know what defaults you may have changed, and its inappropriate for Geany devs to take responsibility when they are advising against doing it.

**Reply by ineuw** [reply]
Thanks. Geany has no responsibility whatsoever, and I understood what you meant.
